src/membership_mesh.rs

```diff
@@ -0,0 +1,125 @@
+//! Rapid mesh-membership based functionality
+use crate::settings::Settings;
+use blip::{Member, Mesh, MeshService, Subscription};
+use chrono::Utc;
+use evmap::{ReadHandle, WriteHandle};
+use log::info;
+use rand::Rng;
+use sled;
+use std::{
+    net::{IpAddr, Ipv6Addr, SocketAddr},
+    thread,
+};
+use tonic::{
+    self,
+    transport::{self, Channel},
+};
+
+/// Attempt to join a mesh if one is available, and if not, starts one up
+/// Returns an EVMap that readers can clone and used to read active members
+pub async fn boostrap(
+    cfg: &Settings,
+) -> anyhow::Result<(String, ReadHandle<MeshKey, Vec<PeerNode>>)> {
+    info!("Attempting to start up Sled database");
+    let db = sled::open(&cfg.data.path)?;
+    info!("Database opened. Was recovered? {}", db.was_recovered());
+    let peer_tree = db.open_tree("peer_configs")?;
+    info!("Peer configuration tree opened");
+    let mem_tree = db.open_tree("members")?;
+    info!("membership data tree opened");
+    let mesh = Mesh::low_latency();
+    info!("Attempting to start low-latency mesh");
+    let listen_addr = acquire_local_address();
+    log::info!("Listening on port: {}", listen_addr.port());
+    let (r_map, mut w_map) = evmap::new();
+    let live_peer_map = PeerMesh { member_map: w_map };
+    let mesh_future = match reconcile_addrs(listen_addr, &cfg.peers.address, None) {
+        StartupMode::Solo => mesh.serve(listen_addr),
+        StartupMode::JoinCluster(peer_addr) => {
+            mesh.join_seed(peer_addr, cfg.use_tls).serve(listen_addr)
+        }
+    };
+    mesh_future.await; // Send this to a specific thread
```

Comment:
We want to execute this mesh/evmap management on a thread, so that if need be, we can treat the membership monitoring like a service in .net and if it fails, or needs to be reconfigured in a way that requires Blips' mesh to be restarted, we can do so, without restarting the whole application.